### ISSUE TITLE
Because of SC1124 I can't ignore SC2016

### ISSUE THREAD

**Issue by SamuelMarks** [koalaman/shellcheck#3120]
#### For bugs
- Rule Id (if any, e.g. SC1000):
- My shellcheck version (`shellcheck --version` or "online"):
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

dollar_par_count=0
case "${1}" in
  '$(')
  # shellcheck disable=SC2003
  dollar_par_count="$(expr "${dollar_par_count}" - 1)" ;;
esac
printf 'dollar_par_count = %d\n' "${dollar_par_count}"
```

#### Here's what shellcheck currently says:
> Expressions don't expand in single quotes, use double quotes for that.

#### Here's what I wanted or expected to see:
Some way of disabling this warning. I'm writing a simple shell parser; following https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html to ensure it's complete… but I also want to use shellcheck.

**Comment by brother** [comment]
And you do not want to have the directive for the complete case for various reasons. If someone want to support directives per case branch.... 

(adding the disable before the case works fine for silencing the error in the example)
